### ISSUE TITLE
Expand upon "Download failed"? 

### ISSUE THREAD

**Issue by sigven** [tidyverse/googledrive#399]
Hi,

Thanks for a very useful package. For reasons I do not fully understand, I experience from time to time download errors when using `googledrive` to download data from some of my R packages, i.e.

`Error in googledrive::drive_download(): ! Download failed.`

I suspect it has to do with some download limitations by the Google Drive API, but I am not sure.

Would it be possible to utilize more information regarding the [request made](https://github.com/tidyverse/googledrive/blob/main/R/drive_download.R#L103-L107), so that the user is somewhat better informed regarding the reason for why the download in fact failed? 

kind regards,
Sigve

**Comment by jennybc** [comment]
To work on this, it's very helpful to have a reliable way to experience the problem (the one leading to `Download failed`). Do you happen to have such a thing? Error handling with these APIs is notoriously hard because it can be hard to get a good example of failure to code against.

**Comment by sigven** [comment]
Totally understand. I basically encounter it now and then, and this is why I find it confusing and wanted to figure out why it sometimes fails. But if this is related to some (to me unclear) restrictions that Google imposes on the user download access frequency (?), I guess it's challenging to find a reliable example that I know will fail. 

A user of my package [just reported an issue] where he experienced a download fail, but I have successfully called the same function (`oncoEnrichR::load_db()`) tonight, so I am fairly unsure.

**Comment by jennybc** [comment]
I can create a branch that actually processes the response in the error case. And then, if you install googledrive from that branch, next time you experience such a failure, you should be able to retrieve the last error from gargle (another package used under the hood). Then you could convey that to me. Do you sometimes see this error yourself?

**Comment by sigven** [comment]
Good idea, that sounds like a very helpful strategy. Yes, as I was doing some repeated testing last night, I saw the same error myself.

**Comment by sigven** [comment]
Great! Thanks a lot for handling this so quickly.
